src/RBDyn/RBDyn/MultiBody.h

```diff
@@ -73,13 +73,21 @@ class RBDYN_DLLAPI MultiBody
   /// @return Body at num position in bodies list.
   const Body & body(int num) const
   {
-    return bodies_[num];
+    return bodies_[static_cast<std::size_t>(num)];
   }
 
   /// Set the body num in bodies list.
   void body(int num, const Body & b)
   {
-    bodies_[num] = b;
+    bodies_[static_cast<std::size_t>(num)] = b;
+  }
+
+  /// @return Index of the root body
+  int rootBodyIndex() const
+  {
+    int index = bodyIndexByName(bodies_[0].name());
+    while(parent(index) != -1) index = parent(index);
+    return index;
```

Comment:
Again, maybe @sonny-tarbouriech had a reason for this. I'll ask him to come and comment on this

Reply:
I was not sure that it would always return 0. But if this is the case we can remove this function.

Reply:
Actually it is guaranteed to be zero if the `MultiBody` was built from a `MultiBodyGraph` but you could construct a `MultiBody` by hand that does not follow that convention. However, all algorithms assume that body 0 is the root of the tree... I think rather than having this function we should check that `parent(0) == -1` when constructing the `MultiBody`